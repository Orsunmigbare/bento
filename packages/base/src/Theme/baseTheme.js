import { ThemeManager } from "@staccx/theme"

export const theme = {
  name: "Default theme",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px",
    micro: "3px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px",
    breakout: "640px"
  },
  color: {
    white: "#fff",
    bg: "#fff",
    subtleHover: "#F7F7FF",
    bgGray: "#f7f7ff",
    line: "#EBEBF2",
    gray: "#767676",
    wcag: "#777",
    disabled: "#BABABA",
    primary: "#00a190",
    secondary: "#F5A623",
    text: "#2b2b2b",
    negative: "#ea526f",
    positive: "#4A90E2",
    warning: "#c54059",
    black: "#000",
    green: "#00a190" // for toggle eks
  },
  fontFamily: {
    heading: "sans-serif",
    body: "sans-serif"
  },
  font: {
    h1: "36px",
    h2: "26px",
    h3: "22px",
    h4: "18px",
    h5: "16px",
    h6: "14px",
    base: "14px",
    tiny: "12px",
    input: "16px",
    huge: "44px"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

const manager = new ThemeManager(theme)
export default manager
