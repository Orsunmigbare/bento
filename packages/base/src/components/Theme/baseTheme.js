export const theme = {
  name: "StaccXBase",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px"
  },
  color: {
    bg: "#fff",
    bgGray: "#f7f7ff",
    primary: "#00a190",
    gray: "#767676",
    text: "#2b2b2b",
    white: "#fff",
    red: "#ea526f",
    green: "#00a190",
    blue: "#988EFB",
    yellow: "#f4da87",
    line: "#EBEBF2",
    wcag: "#777",
    disabled: "#BABABA",
    warning: "#c54059",
    black: "#000",
    subtleHover: "#F7F7FF"
  },
  font: {
    size: {
      h1: "36px",
      h2: "26px",
      h3: "22px",
      h4: "18px",
      h5: "16px",
      h6: "14px",
      base: "14px",
      tiny: "12px",
      input: "16px",
      huge: "44px"
    },
    heading: "Libre Baskerville",
    body: "Libre Franklin"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

export default theme
