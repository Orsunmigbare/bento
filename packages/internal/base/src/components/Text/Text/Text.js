import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import {
  commonPropTypes,
  themePropTypes
} from "../../../constants/themeContants"
import { applyVariants } from "../../../theming"

const Text = ({ level, children, className, ...restProps }) => {
  return (
    <Span className={className} {...restProps}>
      {children}
    </Span>
  )
}

Text.themeProps = {
  text: {
    name: "text",
    description: "",
    type: themePropTypes.style
  }
}

export const TEXT = "text"

export const Span = styled.span`
  ${applyVariants(Text.themeProps.text)};
`

Text.defaultProps = {
  className: null
}

Text.propTypes = {
  children: commonPropTypes.children.isRequired,
  /**
   * Will translate text given context and language.
   * Ignore is not set
   */
  i18n: PropTypes.string,
  className: PropTypes.string
}

export default Text
