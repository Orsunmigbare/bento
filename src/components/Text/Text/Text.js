import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import { applyVariants, commonPropTypes } from "../../../theming"
import themeProps from "./Text.themeProps"

const Text = ({ level, children, className, ...restProps }) => {
  return (
    <Span className={className} {...restProps}>
      {children}
    </Span>
  )
}

export const TEXT = "text"

export const Span = styled.span`
  ${applyVariants(themeProps.text)};
`

Text.defaultProps = {
  className: null
}

Text.propTypes = {
  children: commonPropTypes.children.isRequired,
  /**
   * Will translate text given context and language.
   * Ignore is not set
   */
  i18n: PropTypes.string,
  className: PropTypes.string
}
Text.themeProps = themeProps

export default Text
