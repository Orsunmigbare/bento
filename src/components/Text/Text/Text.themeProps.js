import { themePropTypes, createVariantsFunctionFactory } from "../../../theming"

const themeProps = {
  text: {
    name: "text",
    description: "",
    type: themePropTypes.style
  }
}

export default createVariantsFunctionFactory(themeProps)
