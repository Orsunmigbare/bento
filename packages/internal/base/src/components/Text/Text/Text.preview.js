import React from "react"
import Text from "./Text"

const preview = {
  title: "Text",
  category: "Components/Texts",
  tags: ["typography"],
  component: Text,
  render: props => <Text {...props}>This is a Text</Text>
}

export default preview
