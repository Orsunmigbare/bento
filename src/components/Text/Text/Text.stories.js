import React from "react"
import { storiesOf } from "@storybook/react"
import { withKnobs, text, number } from "@storybook/addon-knobs"
import Text from "./Text"

storiesOf("Components|Base/Text/Text", module)
  .addDecorator(withKnobs)
  .add("no props", () => <Text>This is a text </Text>)

  .add("With className", () => {
    const nu = number("value", 1)
    const className = text("className", "Test")

    const content = `i Am ${className}`
    return (
      <div>
        {content}
        {nu}
      </div>
    )
  })
