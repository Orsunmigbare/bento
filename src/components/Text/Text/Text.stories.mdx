import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import Text from "./Text"

<Meta title="Components|Text/Text" component={Text}/>

# Text
Text is is like a span tag, but it also takes inn i18n (translation) as props

```javascript
import { Text } from "@staccx/bento"
```

## Standard Text


<Preview>
  <Story name="Standard">
   <Text>This is within the Text tag</Text>
  </Story>
</Preview>


## Props for Text
<Props of={Text} />
