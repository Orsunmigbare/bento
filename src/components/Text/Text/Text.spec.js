import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../../theming/themes/baseTheme"
import Text from "./Text"

describe("Text", () => {
  describe("Snapshots", () => {
    it("No props", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Text>This is a Text</Text>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("no Children", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Text />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("ClassName", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Text className="className">hei</Text>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
