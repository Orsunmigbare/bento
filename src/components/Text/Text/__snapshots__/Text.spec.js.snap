// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Snapshots ClassName 1`] = `
<span
  className="sc-AxjAm jzbbdJ className"
>
  hei
</span>
`;

exports[`Text Snapshots No props 1`] = `
<span
  className="sc-AxjAm jzbbdJ"
>
  This is a Text
</span>
`;

exports[`Text Snapshots no Children 1`] = `
<span
  className="sc-AxjAm jzbbdJ"
/>
`;
